include/stdx/algorithm.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <functional>
+#if __cplusplus >= 202002L
+#include <stdx/tuple.hpp>
+#else
+#include <tuple>
+#endif
+
+namespace stdx {
+inline namespace v1 {
+namespace detail {
+#if __cplusplus >= 202002L
+template <typename... Ts> using result_tuple_t = stdx::tuple<Ts...>;
+#define CONSTEXPR_INVOKE constexpr
+#else
+template <typename... Ts> using result_tuple_t = std::tuple<Ts...>;
+#define CONSTEXPR_INVOKE
+#endif
+} // namespace detail
+
+template <typename O, typename... Is>
+using transform_result = detail::result_tuple_t<O, Is...>;
+
+template <typename InputIt, typename OutputIt, typename Operation,
+          typename... InputItN>
+CONSTEXPR_INVOKE auto transform(InputIt first, InputIt last, OutputIt d_first,
+                                Operation op, InputItN... first_n)
+    -> transform_result<OutputIt, InputIt, InputItN...> {
+    while (first != last) {
+        *d_first = std::invoke(op, *first, *first_n...);
```

Comment:
question: why use `std::invoke` instead of the call operator?

Reply:
The usual generic reasons. PMFs, PMDs, etc.